src/euphorie/deployment/browser/sitemenu.py

```diff
@@ -10,9 +10,47 @@
 from plonetheme.nuplone.utils import getPortal
 from Products.CMFCore.utils import getToolByName
 from Products.membrane.interfaces.user import IMembraneUser
+from zope.component import queryMultiAdapter
 
 
 class Sitemenu(sitemenu.Sitemenu):
+    @property
+    def actions(self):
+        """See plonetheme.nuplone.skin.sitemenu.py."""
+        menu = super(Sitemenu, self).actions or {}
+        children = menu.get("children")
+        if not children:
+            return None
+
+        submenu = self.menu_country_tools()
+        if submenu:
+            self.add_submenu(children, submenu)
+        if children:
+            return menu
+        else:
+            return None
+
+    def menu_country_tools(self):
+        context = aq_inner(self.context)
+        request = self.request
+
+        # We try to render the view. It would fail for the wrong context (not
+        # an ICountry) or if permissions are not met.
```

Comment:
I'm pretty sure `queryMultiAdapter` doesn't check permissions. In this case it doesn't matter much because `country-tools` is registered with `zope2.View`. But I would suggest either not mentioning permissions in the comment or double checking that permissions are actually respected (e.g. by writing a test for that).

Reply:
That was basically copying the logic from https://github.com/euphorie/osha.oira/blob/7645c3d5dbe8fc157b02f1c042c5d91bf7b738d8/src/osha/oira/content/sitemenu.py#L31
But I‌ think you're right. I'm checking that.

Reply:
While we're at it, querying the adapter doesn't render the view. Maybe rephrasing that would be good as well.

... and the comment in osha.oira should probably be changed as well. Wow, that's 10 years old...

Reply:
done.